tomcat-8.5/src/org/dogtagpki/tomcat/JSSContext.java

```diff
@@ -0,0 +1,102 @@
+package org.dogtagpki.tomcat;
+
+import java.util.List;
+import java.security.Provider;
+import java.security.Security;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+
+import org.apache.tomcat.util.net.SSLContext;
+
+import org.mozilla.jss.provider.javax.crypto.JSSKeyManager;
+import org.mozilla.jss.ssl.javax.JSSEngine;
+import org.mozilla.jss.ssl.javax.JSSParameters;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JSSContext implements org.apache.tomcat.util.net.SSLContext {
+    public static Logger logger = LoggerFactory.getLogger(JSSContext.class);
+
+    private javax.net.ssl.SSLContext ctx;
+    private JSSParameters params;
+
+    public JSSContext(String alias, List<String> negotiableProtocols) {
+        logger.debug("TomcatJSSContext(" + alias + ", ...");
+
+        params = new JSSParameters();
+        params.setAlias(alias);
+        params.setProtocols(negotiableProtocols.toArray(new String[0]));
+    }
+
+    public void init(javax.net.ssl.KeyManager[] kms, javax.net.ssl.TrustManager[] tms, java.security.SecureRandom sr) throws java.security.KeyManagementException {
```

Comment:
Let's add import statements for these classes.

Reply:
ACK, I can do that. The [Tomcat docs](https://tomcat.apache.org/tomcat-8.5-doc/api/org/apache/tomcat/util/net/SSLContext.html) show full class names for some reason.